toptal/picasso
# feat(picasso): adopt picasso packages for react@18 strict mode

[SP-3419]

### Description

As the preparation for adding `StrictMode` to the StaffPortal ([SPT-2967: Add `StrictMode` to the applicationTO DO](https://toptal-core.atlassian.net/browse/SPT-2967)) we need to migrate picasso from `react-helmet` package (this package uses `react-side-effect` package that uses old lifecycle API) to `react-helmet-async` package [GitHub - staylor/react-helmet-async: Thread-safe Helmet for React 16+ and friends](https://github.com/staylor/react-helmet-async)

Because `Picasso` is also used with `NextJS` and currently we use bundled `Head` component (from `next/head`), to not break anything, the `disableHelmet` prop is added to the `PicassoProvider` component.

⚠️ This prop should be set to `true` on server environment! ⚠️ 

Because of this change, this is the breaking change release.

Also now we use `unstable_createMuiStrictModeTheme` instead of `createTheme` that supports `Strict Mode` in `react 18`
I tested `unstable_createMuiStrictModeTheme` in https://github.com/toptal/keycodes-app and it works fine

### How to test

- CI should pass
- check updated documentation:
- https://picasso.toptal.net/SP-3419-react-helmet-async/?path=/story/tutorials-how-to-use-server-side-rendering--how-to-use-server-side-rendering
- https://picasso.toptal.net/SP-3419-react-helmet-async/?path=/story/layout-page--page
- check SP temploy https://staff-portal-pr-8693.toptal.rocks/. You should not see any visual bugs
- meta tags that are added by `react-helmet-async` should exist in the DOM. See the screenshot
![image](https://user-images.githubusercontent.com/9210534/194700957-11f683da-d376-437c-af5d-bd5b5d40c670.png)


### Development checks

- [n/a] Create `examples` for component
- [n/a] Covered with tests

**Breaking change**

- [n/a] codemod is created and showcased in the changeset

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[SP-3419]: https://toptal-core.atlassian.net/browse/SP-3419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
